gravitee-apim-repository/gravitee-apim-repository-test/src/test/java/io/gravitee/repository/management/EventRepositoryTest.java

```diff
@@ -248,36 +251,114 @@ public void searchByCollectionPropertyWithoutPagingAndBoundary() throws Exceptio
     }
 
     @Test
-    public void searchByEnvironmentDefault() throws Exception {
+    public void searchByEnvironmentDefault() {
         List<Event> events = eventRepository.search(EventCriteria.builder().environments(singletonList("DEFAULT")).build());
 
-        assertEquals(8L, events.size());
-        final Iterator<Event> iterator = events.iterator();
-        assertEquals("event09", iterator.next().getId());
-        assertEquals("event08", iterator.next().getId());
-        assertEquals("event06", iterator.next().getId());
-        assertEquals("event05", iterator.next().getId());
-        assertEquals("event04", iterator.next().getId());
-        assertEquals("event03", iterator.next().getId());
-        assertEquals("event02", iterator.next().getId());
-        assertEquals("event01", iterator.next().getId());
+        assertEquals(12L, events.size());
+        assertThat(events.stream().map(Event::getId))
```

Comment:
Why are there more items, and why has the order been reversed?

Reply:
More events are matching. You're right, I didn't check the order. I've reintroduced the check with order (order by updated at desc) and checked against all the database implementations.
![image](https://github.com/gravitee-io/gravitee-api-management/assets/5955401/dc0b7504-8757-4d93-b19b-0d0d8195da02)